crates/swift-bridge-ir/src/bridged_type.rs

```diff
@@ -440,6 +444,17 @@ impl BridgeableType for BridgedType {
         }
     }
 
+    fn is_passed_via_pointer(&self) -> bool {
+        match self {
+            BridgedType::StdLib(ty) => match ty {
+                StdLibType::Null => true,
```

Comment:
https://github.com/chinedufn/swift-bridge/blob/5acac1d4ba9269ed0de8d583a9c01509084ab197/crates/swift-bridge-ir/src/codegen/codegen_tests/result_codegen_tests.rs#L355-L377

In the above following, as far as I understand, `Null` seems to be passed over FFI via a pointer.

Reply:
`Result<(), OpaqueType>` is bridged via pointer, but when `()` is bridged on its own it is not bridged via pointer.

Maybe we can rename this method to `is_always_bridged_via_pointer` to make this more clear?

---

`ResultVoidAndPtr` is used when you are bridging a type that is passed via pointer (such as `type MyRustType`) with a type that has a zero byte FFI encoding (such as `()` or `struct UnitStruct;`.

When you bridge a `()` on its own it isn't passed via pointer. Actually, it doesn't have any encoding on its own. Because it's a zero-sized type, we can "encode" it using zero bytes.

For a `Result<(), MyRustType>`, the `MyRustType` can be encoded as a non-null pointer, and we can use the null pointer to indicate that it is the `()` variant.

In fact `ResultVoidAndPtr` shouldn't exist at all since we can encode this into a pointer without the `is_ok` (a null pointer optimization). I just didn't catch it in code review https://github.com/chinedufn/swift-bridge/commit/27d7be991141586fb120e4691b7bea7d9eeecd17#r102860388 . (We don't have to fix it in this PR).

---

Here's how `()` is bridged when it's on its own and isn't inside of a `Result<T, E>`.

https://github.com/chinedufn/swift-bridge/blob/c15ba303a2c33dc457ddc2078cd0f8dbdc8e96de/crates/swift-bridge-ir/src/codegen/codegen_tests/single_representation_type_elision_codegen_tests.rs#L10-L82